tests/geocoding_tests.py

```diff
@@ -15,62 +15,323 @@
 
 from .base_tests import SupersetTestCase
 
 
 class GeocodingTests(SupersetTestCase):
-    superset = core.Superset()
+    test_database = None
+    sqla_departments = None
 
     def __init__(self, *args, **kwargs):
         super(GeocodingTests, self).__init__(*args, **kwargs)
 
     def setUp(self):
         self.login()
+        Geocoder.geocoder_util = GeocoderUtilMock(conf)
+        self.create_table_in_view()
 
-    def tearDown(self):
+    def create_table_in_view(self):
+        self.test_database = db.session.query(Database).first()
+        self.test_database.allow_dml = True
+
+        params = {"remote_id": 1234, "database_name": self.test_database.name}
+        self.sqla_departments = SqlaTable(
+            id=1234, table_name="Departments", params=json.dumps(params)
+        )
+        self.sqla_departments.columns.append(
+            TableColumn(column_name="department_id", type="INTEGER")
+        )
+        self.sqla_departments.columns.append(
+            TableColumn(column_name="name", type="STRING")
+        )
+        self.sqla_departments.columns.append(
+            TableColumn(column_name="street", type="STRING")
+        )
+        self.sqla_departments.columns.append(
+            TableColumn(column_name="city", type="STRING")
+        )
+        self.sqla_departments.columns.append(
+            TableColumn(column_name="country", type="STRING")
+        )
+        self.sqla_departments.columns.append(
+            TableColumn(column_name="lat", type="FLOAT")
+        )
+        self.sqla_departments.columns.append(
+            TableColumn(column_name="lon", type="FLOAT")
+        )
+        self.sqla_departments.database = self.test_database
+        self.sqla_departments.database_id = self.sqla_departments.database.id
+        db.session.add(self.sqla_departments)
+        db.session.commit()
+
+        data = {
+            "department_id": [1, 2, 3, 4, 5],
+            "name": [
+                "Logistics",
+                "Marketing",
+                "Facility Management",
+                "Personal",
+                "Finances",
+            ],
+            "street": [
+                "Oberseestrasse 10",
+                "Grossmünsterplatz",
+                "Uetliberg",
+                "Zürichbergstrasse 221",
+                "Bahnhofstrasse",
+            ],
+            "city": ["Rapperswil", "Zürich", "Zürich", "Zürich", "Zürich"],
+            "country": [
+                "Switzerland",
+                "Switzerland",
+                "Switzerland",
+                "Switzerland",
+                "Switzerland",
+            ],
+            "lat": [None, None, None, None, None],
+            "lon": [None, None, None, None, None],
+        }
+        df = pd.DataFrame(data=data)
+
+        # because of caching problem with postgres load database a second time
+        # without this, the sqla engine throws an exception
+        database = db.session.query(Database).first()
+        if database:
```

Comment:
What if we fail to get the database? then the table will have no entries. Doesn't this change the tests?